### PR TITLE
/api/beneficiary/makeAnnouncements route

### PR DESCRIPTION
**Pull request by lucashu1**
Use templated messages to send messages to all our (visible) beneficiary families.

Sample request body:

```
{
	"messageTemplates": {
		"en": "hello world! {lastName}",
		"fa": "سلام! {lastName}",
		"ar": "مرحبا! {lastName}"
	}
}
```

Sample messages (all sent to me bc all `fb_psid`s in sandbox are just my personal `fb_psid`):
Screen Shot 2019-12-04 at 9 09 00 PM

When sending announcements, all messages will be tested on `staging` first to make sure that the RTL text appears correctly, and all templated fields are populated correctly.

### CODE REVIEW

File: src/server/controller/beneficiary.js [hunk 1]
```diff
@@ -53,8 +54,26 @@ async function getBeneficiaryScores(req, res) {
   }
 }
 
+async function makeFBAnnouncementToVisibleBeneficiaries(req, res) {
+  try {
+    const messageTemplates = req.body.messageTemplates;
```
Comment (QuinnNgo) [inline src/server/controller/beneficiary.js]:
```suggestion
    const { messageTemplates } = req.body;
```

File: src/server/controller/beneficiary.js [hunk 2]
```diff
@@ -53,8 +54,26 @@ async function getBeneficiaryScores(req, res) {
...
+    // check to make sure all languages are accounted for
+    if (!messageTemplates.en || !messageTemplates.fa || !messageTemplates.ar) {
+      console.log(`Tried to make FB announcement, but not all languages were included! messageTemplates: ${JSON.stringify(messageTemplates)}`);
+      return res.status(400).send();
```
Comment (QuinnNgo) [inline src/server/controller/beneficiary.js]:
```suggestion
      return res.sendStatus(400);
```

File: src/server/controller/beneficiary.js [hunk 3]
```diff
@@ -53,8 +54,26 @@ async function getBeneficiaryScores(req, res) {
...
+    }
+    await fbHelpers.sendFBMessageToAllVisibleBeneficiaries(messageTemplates);
+    console.log(`Successfully made FB announcement using messageTemplates: ${JSON.stringify(messageTemplates)}`);
+    return res.status(200).send();
```
Comment (QuinnNgo) [inline src/server/controller/beneficiary.js]:
```suggestion
    return res.sendStatus(200);
```

File: src/server/controller/beneficiary.js [hunk 4]
```diff
@@ -53,8 +54,26 @@ async function getBeneficiaryScores(req, res) {
...
+  } catch (err) {
+    errorHandler.handleError(err, "beneficiary/makeFBAnnouncementToVisibleBeneficiaries");
+    return res.status(500).send();
```
Comment (QuinnNgo) [inline src/server/controller/beneficiary.js]:
```suggestion
    return res.sendStatus(500);
```

File: src/server/util/beneficiaryHelpers.js [hunk 5]
```diff
@@ -48,6 +49,19 @@ async function getBeneficiaryObjWithoutNeedsFromBeneficiaryId(beneficiaryId) {
   }
 }
 
+async function getAllBeneficiaryObjsWithoutNeeds() {
+  try {
+    const conn = await config.dbInitConnectPromise();
+    const [rows, fields] = await conn.query(
```
Comment (QuinnNgo) [inline src/server/util/beneficiaryHelpers.js]:
```suggestion
    const [rows] = await conn.query(
```

File: src/server/util/fbHelpers.js [hunk 6]
```diff
@@ -196,5 +219,8 @@ export default {
   sendSuccessfulItemRequestMessageNoBudget,
 
   // Pickup notifications
-  sendPickupNotification
+  sendPickupNotification,
+
+  // Announcements
+  sendFBMessageToAllVisibleBeneficiaries
 }
```
Comment (Raghav-Maheshwari) [inline src/server/util/fbHelpers.js]: add newline

**Pull request merged**